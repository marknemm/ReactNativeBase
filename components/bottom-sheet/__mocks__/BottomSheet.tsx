import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
