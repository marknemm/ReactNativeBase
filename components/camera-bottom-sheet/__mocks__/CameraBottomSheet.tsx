import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
