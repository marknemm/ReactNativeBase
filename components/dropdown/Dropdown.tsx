import ErrorText from '@components/error-text/ErrorText';
import { useFormControl, useFormErrorMessage, useValidationRules } from '@hooks/form-hooks';
import { Text, useTheme } from '@rneui/themed';
import { forwardRef } from 'react';
import { Controller } from 'react-hook-form';
import { View } from 'react-native';
import { Dropdown as RneDropdown } from 'react-native-element-dropdown';
import { useDropdownItems } from './Dropdown.hooks';
import type { DropdownFC, DropdownProps } from './Dropdown.interfaces';
import { useStyles } from './Dropdown.styles';

const DEFAULT_LABEL_FIELD = 'label';
const DEFAULT_VALUE_FIELD = 'value';

/**
 * A component for a dropdown form field.
 *
 * @param props The component {@link DropdownProps}.
 * @param ref The component reference.
 * @returns The {@link Dropdown} component.
 */
const Dropdown: DropdownFC = forwardRef((props, ref) => {
  const { label, name, onBlur, onChange, placeholder, valueField = DEFAULT_VALUE_FIELD } = props;

  // Derive entities related to a dropdown controlled by react-hook-form
  const control = useFormControl(props);
  const rules = useValidationRules(props, label?.toString() || placeholder);
  const errorMessage = useFormErrorMessage(rules, props);

  return control
    ? (
      <>
        <Controller
          control={control}
          name={name}
          render={({ field: { onBlur: onBlurForm, onChange: onChangeForm, value } }) => (
            <DropdownControlled
              {...props}
              onBlur={() => {
                onBlurForm();
                onBlur?.();
              }}
              onChange={(item) => {
                onChangeForm(item?.[valueField]);
                onChange?.(item);
              }}
              ref={ref as any}
              value={value}
            />
          )}
          rules={rules}
        />

        <ErrorText
          {...props}
          error={errorMessage}
          style={null}
        />
      </>
    )
    : <DropdownControlled {...props} ref={ref as any} />;
});

/**
 * The controlled {@link Dropdown} component.
 *
 * @param props The component {@link DropdownProps}.
 * @param ref The component reference.
 * @returns The {@link DropdownControlled} component.
 */
const DropdownControlled: DropdownFC = forwardRef(({
  containerStyle,
  data,
  includeEmptyOption = false,
  itemContainerStyle,
  itemTextStyle,
  label,
  labelField = DEFAULT_LABEL_FIELD,
  labelStyle,
  onChange,
  placeholderStyle,
  selectedTextStyle,
  style,
  valueField = DEFAULT_VALUE_FIELD,
  ...rneDropdownProps
}, ref) => {
  const styles = useStyles({
    containerStyle,
    itemContainerStyle,
    itemTextStyle,
    labelStyle,
    placeholderStyle,
    selectedTextStyle,
    style,
  });
  const { theme } = useTheme();
  const items = useDropdownItems(data, labelField, valueField, includeEmptyOption);

  return (
    <View style={styles.container}>
      {label && (
        <Text style={styles.label}>
          {label}
        </Text>
      )}

      <RneDropdown
        activeColor={theme.colors.grey5}
        data={items}
        itemContainerStyle={styles.itemContainer}
        itemTextStyle={styles.itemText}
        labelField={labelField}
        onChange={onChange}
        placeholderStyle={styles.placeholder}
        ref={ref as any}
        selectedTextStyle={styles.selectedText}
        style={styles.dropdown}
        valueField={valueField}
        {...rneDropdownProps}
      />
    </View>
  );
});

export type * from './Dropdown.interfaces';
export default Dropdown;
