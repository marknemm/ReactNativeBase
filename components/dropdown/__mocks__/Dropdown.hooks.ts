/**
 * Mocks the useDropdownItems hook.
 */
export const useDropdownItems = jest.fn().mockImplementation((data) => data);
