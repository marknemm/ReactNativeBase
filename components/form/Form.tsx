import { FormProvider } from 'react-hook-form';
import { View } from 'react-native';
import type { FormProps } from './Form.interface';

/**
 * A component that wraps a form and provides form context to its children.
 *
 * @param props The component {@link FormProps}.
 * @returns The {@link Form} component.
 */
const Form: React.FC<FormProps> = ({ children, form, style, ...restViewProps }) => (
  <FormProvider {...form}>
    <View
      style={style}
      testID="rnb-form-view"
      {...restViewProps}
    >
      {children}
    </View>
  </FormProvider>
);

export type * from './Form.interface';
export default Form;
