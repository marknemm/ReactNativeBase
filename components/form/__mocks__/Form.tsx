import type { FormProps } from '@components/form/Form.interface';

const FormMock = jest.fn().mockImplementation((props: FormProps) => {
  const ActualComponent = jest.requireActual('@components/form/Form').default;
  return <ActualComponent {...props} />;
});

export default FormMock;
