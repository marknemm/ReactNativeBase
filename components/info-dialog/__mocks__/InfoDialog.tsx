import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
