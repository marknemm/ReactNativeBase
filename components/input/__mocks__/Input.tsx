import type { InputProps } from '@components/input/Input.interfaces';

const MockInput = jest.fn().mockImplementation((props: InputProps) => {
  const ActualComponent = jest.requireActual('@components/input/Input').default;
  return <ActualComponent {...props} />;
});

export default MockInput;
