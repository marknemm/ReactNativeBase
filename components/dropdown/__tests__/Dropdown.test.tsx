import Dropdown from '@components/dropdown/Dropdown';
import { useDropdownItems } from '@components/dropdown/Dropdown.hooks';
import Form from '@components/form/Form';
import { genMockForm } from '@hooks/__mocks__/form-hooks';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen, userEvent } from '@testing-library/react-native';

jest.mock('@components/dropdown/Dropdown.hooks');

describe('<Dropdown />', () => {
  let mockData: { label: string, labelAlt: string, value: string, valueAlt: string }[];

  beforeEach(() => {
    mockData = [
      { label: 'item 1', labelAlt: 'item 1 alt', value: 'value 1', valueAlt: 'value 1 alt' },
      { label: 'item 2', labelAlt: 'item 2 alt', value: 'value 2', valueAlt: 'value 2 alt' },
      { label: 'item 3', labelAlt: 'item 3 alt', value: 'value 3', valueAlt: 'value 3 alt' },
    ];
  });

  describe('dropdown items', () => {
    it('generates dropdown items from the raw data and default options', () => {
      render(
        <Dropdown data={mockData} />,
        { wrapper: AppProvider }
      );

      expect(useDropdownItems).toHaveBeenCalledWith(mockData, 'label', 'value', false);
    });

    it('generates dropdown items from the raw data and custom options', () => {
      render(
        <Dropdown
          data={mockData}
          includeEmptyOption
          labelField="labelAlt"
          valueField="valueAlt"
        />,
        { wrapper: AppProvider }
      );

      expect(useDropdownItems).toHaveBeenCalledWith(mockData, 'labelAlt', 'valueAlt', true);
    });
  });

  describe('value changes', () => {
    it('calls the `onChange` callback when dropdown item is selected', async () => {
      const onChange = jest.fn();
      render(
        <Dropdown
          data={mockData}
          onChange={onChange}
        />,
        { wrapper: AppProvider }
      );

      await openDropdown();
      const dropdownItem = screen.getByText(mockData[0].label);
      await userEvent.press(dropdownItem);

      expect(onChange).toHaveBeenCalledWith(expect.objectContaining(mockData[0]));
    });
  });

  describe('form control', () => {
    it('Updates the form value and calls the `onChange` callback when dropdown item is selected', async () => {
      const form = genMockForm({
        dropdown: '',
      });
      const onChange = jest.fn();

      render(
        <Form form={form}>
          <Dropdown
            data={mockData}
            name="dropdown"
            onChange={onChange}
          />
        </Form>,
        { wrapper: AppProvider }
      );

      await openDropdown();
      const dropdownItem = screen.getByText(mockData[1].label);
      await userEvent.press(dropdownItem);

      expect(form.getValues('dropdown')).toBe(mockData[1].value);
      expect(onChange).toHaveBeenCalledWith(expect.objectContaining(mockData[1]));
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <Dropdown data={mockData} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with label and placeholder', () => {
      render(
        <Dropdown
          data={mockData}
          label="Test Dropdown Label"
          placeholder="Test Dropdown Placeholder"
        />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when open', async () => {
      render(
        <Dropdown data={mockData} />,
        { wrapper: AppProvider }
      );

      await openDropdown();
      expect(screen.toJSON()).toMatchSnapshot();
    });
  });

  it('renders correctly when open with alternate label and value fields', async () => {
    render(
      <Dropdown
        data={mockData}
        labelField="labelAlt"
        valueField="valueAlt"
      />,
      { wrapper: AppProvider }
    );

    await openDropdown({ labelField: 'labelAlt' });
    expect(screen.toJSON()).toMatchSnapshot();
  });

  async function openDropdown({
    placeholder = 'Select item',
    labelField = 'label' as keyof typeof mockData[0],
  } = {}) {
    const dropdownToggle = screen.getByText(placeholder);
    await userEvent.press(dropdownToggle);
    await screen.findByText(mockData[0][labelField]);
  }
});
