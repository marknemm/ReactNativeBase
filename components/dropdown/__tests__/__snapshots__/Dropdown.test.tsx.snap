// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly when open with alternate label and value fields 1`] = `
<View
  style={
    {
      "marginBottom": 6.285714285714286,
      "paddingHorizontal": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "height": 98.9730612244898,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "fontSize": 16,
              "writingDirection": "ltr",
            },
            {
              "color": "#86939e",
              "fontSize": 23.57142857142857,
            },
            {},
          ]
        }
      >
        Select item
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 20,
            "tintColor": "gray",
            "width": 20,
          }
        }
      />
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      statusBarTranslucent={true}
      supportedOrientations={
        [
          "landscape",
          "portrait",
        ]
      }
      transparent={true}
      visible={true}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingTop": 582,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "white",
                "borderColor": "#EEEEEE",
                "borderWidth": 0.5,
                "elevation": 2,
                "flexShrink": 1,
                "left": 0,
                "maxHeight": 340,
                "minHeight": 0,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 1.41,
                "width": 360,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "flexShrink": 1,
                }
              }
            >
              <RCTScrollView
                accessibilityLabel="undefined flatlist"
                data={
                  [
                    {
                      "_index": 0,
                      "label": "item 1",
                      "labelAlt": "item 1 alt",
                      "value": "value 1",
                      "valueAlt": "value 1 alt",
                    },
                    {
                      "_index": 1,
                      "label": "item 2",
                      "labelAlt": "item 2 alt",
                      "value": "value 2",
                      "valueAlt": "value 2 alt",
                    },
                    {
                      "_index": 2,
                      "label": "item 3",
                      "labelAlt": "item 3 alt",
                      "value": "value 3",
                      "valueAlt": "value 3 alt",
                    },
                  ]
                }
                getItem={[Function]}
                getItemCount={[Function]}
                invertStickyHeaders={false}
                inverted={false}
                isInvertedVirtualizedList={false}
                keyExtractor={[Function]}
                keyboardShouldPersistTaps="handled"
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                onScrollToIndexFailed={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={0.0001}
                showsVerticalScrollIndicator={true}
                stickyHeaderIndices={[]}
                testID="undefined flatlist"
                viewabilityConfigCallbackPairs={[]}
              >
                <View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 1 alt"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 1 alt"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 1 alt
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 2 alt"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 2 alt"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 2 alt
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 3 alt"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 3 alt"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 3 alt
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;

exports[`<Dropdown /> snapshots renders correctly 1`] = `
<View
  style={
    {
      "marginBottom": 6.285714285714286,
      "paddingHorizontal": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "height": 98.9730612244898,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "fontSize": 16,
              "writingDirection": "ltr",
            },
            {
              "color": "#86939e",
              "fontSize": 23.57142857142857,
            },
            {},
          ]
        }
      >
        Select item
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 20,
            "tintColor": "gray",
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Dropdown /> snapshots renders correctly when open 1`] = `
<View
  style={
    {
      "marginBottom": 6.285714285714286,
      "paddingHorizontal": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "height": 98.9730612244898,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "fontSize": 16,
              "writingDirection": "ltr",
            },
            {
              "color": "#86939e",
              "fontSize": 23.57142857142857,
            },
            {},
          ]
        }
      >
        Select item
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 20,
            "tintColor": "gray",
            "width": 20,
          }
        }
      />
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      statusBarTranslucent={true}
      supportedOrientations={
        [
          "landscape",
          "portrait",
        ]
      }
      transparent={true}
      visible={true}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingTop": 582,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "white",
                "borderColor": "#EEEEEE",
                "borderWidth": 0.5,
                "elevation": 2,
                "flexShrink": 1,
                "left": 0,
                "maxHeight": 340,
                "minHeight": 0,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 1.41,
                "width": 360,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "flexShrink": 1,
                }
              }
            >
              <RCTScrollView
                accessibilityLabel="undefined flatlist"
                data={
                  [
                    {
                      "_index": 0,
                      "label": "item 1",
                      "labelAlt": "item 1 alt",
                      "value": "value 1",
                      "valueAlt": "value 1 alt",
                    },
                    {
                      "_index": 1,
                      "label": "item 2",
                      "labelAlt": "item 2 alt",
                      "value": "value 2",
                      "valueAlt": "value 2 alt",
                    },
                    {
                      "_index": 2,
                      "label": "item 3",
                      "labelAlt": "item 3 alt",
                      "value": "value 3",
                      "valueAlt": "value 3 alt",
                    },
                  ]
                }
                getItem={[Function]}
                getItemCount={[Function]}
                invertStickyHeaders={false}
                inverted={false}
                isInvertedVirtualizedList={false}
                keyExtractor={[Function]}
                keyboardShouldPersistTaps="handled"
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                onScrollToIndexFailed={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={0.0001}
                showsVerticalScrollIndicator={true}
                stickyHeaderIndices={[]}
                testID="undefined flatlist"
                viewabilityConfigCallbackPairs={[]}
              >
                <View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 1"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 1"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 1
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 2"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 2"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 2
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onFocusCapture={[Function]}
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessibilityLabel="item 3"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="item 3"
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "padding": 17,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#242424",
                                "flex": 1,
                                "fontSize": 16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            item 3
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;

exports[`<Dropdown /> snapshots renders correctly with label and placeholder 1`] = `
<View
  style={
    {
      "marginBottom": 6.285714285714286,
      "paddingHorizontal": 0,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#86939e",
        "fontFamily": undefined,
        "fontSize": 23.57142857142857,
        "fontStyle": "normal",
        "fontWeight": "bold",
        "marginBottom": 0,
        "textAlign": "left",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
      }
    }
  >
    Test Dropdown Label
  </Text>
  <View
    onLayout={[Function]}
    style={
      {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "height": 98.9730612244898,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "fontSize": 16,
              "writingDirection": "ltr",
            },
            {
              "color": "#86939e",
              "fontSize": 23.57142857142857,
            },
            {},
          ]
        }
      >
        Test Dropdown Placeholder
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 20,
            "tintColor": "gray",
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
